src/ddscxx/include/dds/sub/detail/SamplesHolder.hpp

```diff
@@ -64,9 +64,6 @@ class LoanedSamplesHolder : public SamplesHolder
 
         uint32_t cpp_sample_size = this->samples_.delegate()->length();
         void **c_sample_pointers = new void * [cpp_sample_size];
-        for (uint32_t i = 0; i < cpp_sample_size; ++i) {
```

Comment:
Is this correctly deleted?
Since now you just get an empty array of pointer back from this function, which have not even been initialized.

Reply:
That is correct. The SampleInfo is now copied in another loop, that is also used to copy the samples themselves. (See set_sample_contents). It seemed to make sense to copy the Sample and its SampleInfo in the same loop instead of making it two separate loops.